### PR TITLE
API controller - checkpoint 6

### PR DESCRIPTION
**Pull request by brendanmcgivern**
@droberts84
- created `controllers/api/events_controller` to allow registered applications to submit events.
- configured routes for api controller

### CODE REVIEW

File: app/controllers/api/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    if registered_application
+      event = registered_application.events.build(event_params)
+      if event.save
+        render json: event, status: :created
+      else
+        render event.errors, status: :unprocessable_entity
+      end
+    else
+      render json: "Unregistered application", status: :unprocessable_entity
```
Comment (droberts84) [inline app/controllers/api/events_controller.rb]: Great validation / error handling!

Pro Tip, respond with valid JSON key / value pairs.. something like `{error_type: 'validation', message: 'unregistered application'}`

File: app/controllers/api/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def create
+    registered_application = RegisteredApplication.find_by(url: request.env['HTTP_ORIGIN'])
+
+    if registered_application
+      event = registered_application.events.build(event_params)
+      if event.save
+        render json: event, status: :created
+      else
+        render event.errors, status: :unprocessable_entity
```
Comment (droberts84) [inline app/controllers/api/events_controller.rb]: Did you test this line?  Something doesn't look right.  You could also consider adding a method to combine with response below

File: app/views/registered_applications/show.html.erb [hunk 3]
```diff
@@ -17,3 +17,7 @@
 </h1>
 <p><%= link_to @app.url, '#' %></p>
 <br />
+
+<% @events.each do |event, count| %>
```
Comment (droberts84) [inline app/views/registered_applications/show.html.erb]: I assume `@events` is an ActiveRecord rel... What is being fed to `count`?  Did you mean `@events.each do |event|`?

Comment (brendanmcgivern) [inline app/views/registered_applications/show.html.erb]: `@events = @app.events.group_by(&:name)`
This returns an array with the group name at index 0 and an array of all the elements in that group at index 1.

I will change it to:
```
<% @events.each do |event| %>
  <h4><%= "#{event[0]} - #{event[1].count}" %></h4>
<% end %>
```

Comment (droberts84) [inline app/views/registered_applications/show.html.erb]: Ah, in that case your original way is definitely better.  You are using
[Enumberable#group_by](http://devdocs.io/ruby/enumerable#method-i-group_by)
which iterates though a Hash, but from a glance I thought you were thought
you were calling [group on a Rel](
http://devdocs.io/rails/activerecord/querymethods#method-i-group) which
return a Rel.

**Pull request merged**